linode/data_source_linode_images.go

```diff
@@ -0,0 +1,88 @@
+package linode
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/linode/linodego"
+
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+)
+
+func dataSourceLinodeImages() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceLinodeImagesRead,
+		Schema: map[string]*schema.Schema{
+			"filter": filterSchema(),
+			"image": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     dataSourceLinodeImage(),
+			},
+		},
+	}
+}
+
+func dataSourceLinodeImagesRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ProviderMeta).Client
+
+	filter, err := constructFilterString(d, imageValueToFilterType)
+	if err != nil {
+		return fmt.Errorf("failed to construct filter: %s", err)
+	}
+
+	images, err := client.ListImages(context.Background(), &linodego.ListOptions{
+		Filter: filter,
+	})
+
+	if err != nil {
+		return fmt.Errorf("failed to list linode images: %s", err)
+	}
+
+	imagesFlattened := make([]interface{}, len(images))
+	for i, image := range images {
+		imagesFlattened[i] = flattenLinodeImage(&image)
+	}
+
+	d.SetId(filter)
+	d.Set("image", imagesFlattened)
+
+	return nil
+}
+
+func flattenLinodeImage(image *linodego.Image) map[string]interface{} {
```

Comment:
could this be reused?

Reply:
Unfortunately in the image data source it writes directly to the `ResourceData`, so it doesn't seem like we can reuse it there.

Reply:
I meant can we put this function in the resource and reuse it in the data source like with the other types?